src/04-modality-specific-files/11-functional-near-infrared-spectroscopy.md

```diff
@@ -0,0 +1,400 @@
+| NIRSCWHBR                   | Continuous wave deoxygenated hemoglobin (deoxyhemoglobin) concentration measurements. Equivalent to dataTypeLabel HbR in SNIRF.                                              |
+| NIRSCWMUA                   | Continuous wave optical absorption  measurements. Equivalent to dataTypeLabel mua in SNIRF.                                                                                  |
+| ACCEL                       | Accelerometer channel, one channel for each orientation. An extra column `component` for the axis of the orientation MUST be added to the `_*channels.tsv` file (x, y or z). |
+| GYRO                        | Gyrometer channel, one channel for each orientation. An extra column `component` for the axis of the orientation MUST be added to the `_*channels.tsv` file (x, y or z).     |
+| MAGN                        | Magnetomenter channel, one channel for each orientation. An extra column `component` for the axis of the orientation MUST be added to the `_*channels.tsv` file (x, y or z). |
+| MISC                        | Miscellaneous                                                                                                                                                                |
+
+Example:
+
+```Text
+Name         type                   source      detector      wavelength_nominal   units
+S1-D1        NIRSCWAMPLITUDE        A1          Fz            760                  V
+S1-D1        NIRSCWAMPLITUDE        A1          Fz            850                  V
+S1-D2        NIRSCWAMPLITUDE        A1          Cz            760                  V
+S2-D1        NIRSCWAMPLITUDE        A2          Fz            760                  V
+S3-D4        NIRSCWAMPLITUDE        VisS2       VisD4         760                  V
+```
+
+## Optode description (`*_optodes.tsv`)
+
+{{ MACROS___make_filename_template(datatypes=["nirs"], suffixes=["optodes"]) }}
+
+File that provides the location and type of optodes. Note that coordinates MUST be
+expressed  in Cartesian coordinates according to the NIRSCoordinateSystem and
+NIRSCoordinateSystemUnits fields in `*_coordsystem.json`. If an `*_optodes.tsv`
+file is specified, a `*_coordsystem.json` file MUST be specified as well.
+The order of the required columns in the `*_optodes.tsv` file MUST be as listed below.
+
+The x, y, and z positions are for measured locations, for example, with a polhemus
+digitizer. If you also have idealised positions, where you wish the optodes to be
+placed, these can be listed in the template values. SNIRF contains arrays for both
+the 3D and 2D locations of data. The BIDS format MUST store the 3D locations if
+available, and only the 2D locations if 3D positions are unavailable. The storage
+of 2D locations would be indicated by the z field containing an `n/a` value.
```

Comment:
```suggestion
placed, these can be listed in the template values
(for example for "template positions" computed on a sphere).
SNIRF contains arrays for both
the 3D and 2D locations of data. The BIDS format MUST store the 3D locations if
available, and only the 2D locations if 3D positions are unavailable. The storage
of 2D locations would be indicated by the z field containing an `n/a` value.
```

for a resource on computing template positions a sphere, I have shared: https://stefanappelhoff.com/eeg_positions/ ... but we don't need to add it here, because it's mostly for EEG.